### PR TITLE
Navbar HOC #18

### PR DESCRIPTION
**Pull request by codewithkushagra**
Made responsive Navbar 

Laptop view
Screenshot 2023-02-20 at 12 03 45 AM

Phone view
Screenshot 2023-02-20 at 12 04 23 AM

### CODE REVIEW

**Comment by codewithkushagra** [comment]
@vinayak-trivedi and @vvaibhavdesai please review my PR

File: components/Layout/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                            Sign In
+                            <Image src={GITHUB_LOGO} alt="git" />
+                        </button>
+                    </Link>
+                </div>
+            </div>
+            {/* nav for width greater then lg */}
+            <ul className={`lg:flex flex-row ${toggle?'bg-white text-[#1d1283] py-2 shadow-[#334155] shadow-2xl':'hidden'} justify-between w-full`}>
+                <div className="lg:flex flex-row">
+                    <li className="px-4 py-2 lg:flex items-center hidden">
```
Comment (vvaibhavdesai) [inline components/Layout/Navbar.tsx:42]: can we please take the repeating `li` tags out and make them a `map` assign them to a variable and use them here?

File: components/Layout/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    <li className="flex items-center m-2 px-7 py-2 lg:px-5 lg:py-5 font-bold"><Link href="#home">Home</Link></li>
+                    <li className="flex items-center m-2 px-7 lg:px-5 py-2 lg:py-5 font-bold" ><Link href="#welcome">Welcome</Link></li>
+                    <li className="flex items-center m-2 px-7 lg:px-5 py-2 lg:py-5 font-bold" ><Link href="#event">Events</Link></li>
+                    <li className="flex items-center m-2 px-7 lg:px-5 py-2 lg:py-5 font-bold" ><Link href="#member">Members</Link></li>
+                    <li className="flex items-center m-2 px-7 lg:px-5 py-2 lg:py-5 font-bold" ><Link href="#crypto">Crypto</Link></li>
+                    <li className="flex items-center m-2 px-7 lg:px-5 py-2 lg:py-5 font-bold" ><Link href="#status">Status</Link></li>
+                </div>
+                <div className="lg:flex flex-row hidden">
+                    <li className="flex items-center m-2 px-5 py-3">
+                        <Link href="#">
```
Comment (vvaibhavdesai) [inline components/Layout/Navbar.tsx:56]: remove the `Link` tag

**Comment by codewithkushagra** [comment]
@vvaibhavdesai 
Okay, understood 
I have gone through all the comments. Thank you for a thorough review. I will work on these issues and report by 21st Feb, 11:00 PM.

**Comment by codewithkushagra** [comment]
I have resolved all the issues please review once @vvaibhavdesai  and @vinayak-trivedi

**Comment by satyam73** [comment]
> I have resolved all the issues please review once @vvaibhavdesai and @vinayak-trivedi

Try resolving the conversation once you worked on that change.

File: components/Layout/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import { useState } from 'react';
+import Link from 'next/link'
+import REALDEVSQUAD_IMAGE from "public/images/Real-Dev-Squad@1x.svg"
+import GITHUB_LOGO from "public/icons/Github_Logo.svg"
+
+
+const Navbar = () => {
+
+    // maintains hamburger toggle state
+    const [toggle, setToggle] = useState(false)
```
[the remark this answers is not in the document]
Comment (codewithkushagra) [reply components/Layout/Navbar.tsx]: Done

File: components/Layout/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    // Handles toggle of hamburger
+    const onClickHandler = () => {
+        console.log(toggle)
```
[the remark this answers is not in the document]
Comment (codewithkushagra) [reply components/Layout/Navbar.tsx]: Removed

File: components/Layout/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                    <span className="block w-8 h-1 bg-white"></span>
+                    <span className="block w-8 h-1 bg-white"></span>
+                    <span className="block w-8 h-1 bg-white"></span>
+                </div>
+                {/* Sign in with Github */}
+                <div className="flex items-center m-2 px-5 py-3">
+                    <Link href="#">
+                        <button className="border-2 rounded-lg text-sm p-1 gap-1 flex flex-row">
+                            Sign In
+                            <Image src={GITHUB_LOGO} alt="git" />
```
[the remark this answers is not in the document]
Comment (codewithkushagra) [reply components/Layout/Navbar.tsx]: No, I forgot to, I have give the width and height now

File: components/Layout/Navbar.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            <div className="flex flex-row lg:hidden justify-between w-full">
+                {/* hamburger menu */}
+                <div className="m-2 px-5 py-4 space-y-2" onClick={onClickHandler}>
+                    <span className="block w-8 h-1 bg-white"></span>
+                    <span className="block w-8 h-1 bg-white"></span>
+                    <span className="block w-8 h-1 bg-white"></span>
+                </div>
+                {/* Sign in with Github */}
+                <div className="flex items-center m-2 px-5 py-3">
+                    <Link href="#">
```
[the remark this answers is not in the document]
Comment (codewithkushagra) [reply components/Layout/Navbar.tsx]: Removed

File: components/Layout/Navbar.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+                        <button className="border-2 rounded-lg text-sm p-1 gap-1 flex flex-row">
+                            Sign In
+                            <Image src={GITHUB_LOGO} alt="git" />
+                        </button>
+                    </Link>
+                </div>
+            </div>
+            {/* nav for width greater then lg */}
+            <ul className={`lg:flex flex-row ${toggle?'bg-white text-[#1d1283] py-2 shadow-[#334155] shadow-2xl':'hidden'} justify-between w-full`}>
```
[the remark this answers is not in the document]
Comment (codewithkushagra) [reply components/Layout/Navbar.tsx]: okay, I have separated this as a constant now

File: components/Layout/Navbar.tsx (hunk 1 above)
Comment (codewithkushagra) [reply components/Layout/Navbar.tsx:47]: Yeah, I have implemented it this way now

File: components/Layout/Navbar.tsx (hunk 2 above)
Comment (codewithkushagra) [reply components/Layout/Navbar.tsx]: removed

**Pull request closed without merging**